crates/services/txpool_v2/src/storage/graph.rs

```diff
@@ -0,0 +1,477 @@
+                        };
+                        let output =
+                            &node.transaction.outputs()[utxo_id.output_index() as usize];
+                        Self::check_if_coin_input_can_spend_output(output, input)?;
+                        pool_dependencies.push(*node_id);
+                    } else if utxo_validation {
+                        let Some(coin) = persistent_storage
+                            .utxo(utxo_id)
+                            .map_err(|e| Error::Database(format!("{:?}", e)))?
+                        else {
+                            return Err(Error::UtxoNotFound(*utxo_id));
+                        };
+                        if !coin.matches_input(input).expect("The input is coin above") {
+                            return Err(Error::NotInsertedIoCoinMismatch);
+                        }
+                    }
+                }
+                Input::MessageCoinSigned(MessageCoinSigned { nonce, .. })
+                | Input::MessageCoinPredicate(MessageCoinPredicate { nonce, .. })
+                | Input::MessageDataSigned(MessageDataSigned { nonce, .. })
+                | Input::MessageDataPredicate(MessageDataPredicate { nonce, .. }) => {
+                    // since message id is derived, we don't need to double check all the fields
+                    // Maybe this should be on an other function as it's not a dependency finder but just a test
+                    if utxo_validation {
+                        if let Some(db_message) = persistent_storage
+                            .message(nonce)
+                            .map_err(|e| Error::Database(format!("{:?}", e)))?
+                        {
+                            // verify message id integrity
+                            if !db_message
+                                .matches_input(input)
+                                .expect("Input is a message above")
+                            {
+                                return Err(Error::NotInsertedIoMessageMismatch);
+                            }
+                        } else {
+                            return Err(Error::NotInsertedInputMessageUnknown(*nonce));
+                        }
+                    }
+                }
+                Input::Contract(Contract { contract_id, .. }) => {
+                    if let Some(node_id) = self.contracts_creators.get(contract_id) {
+                        pool_dependencies.push(*node_id);
+                    } else if !persistent_storage
+                        .contract_exist(contract_id)
+                        .map_err(|e| Error::Database(format!("{:?}", e)))?
+                    {
+                        return Err(Error::NotInsertedInputContractDoesNotExist(
+                            *contract_id,
+                        ));
+                    }
+                }
+            }
+        }
+        Ok(pool_dependencies)
+    }
+
+    fn remove_transaction(
```

Comment:
It is possible to call this function for any transaction in the graph, but it doesn't update cumulative information for parents.

It looks like we call this function only during selection, so maybe we need to rename it like `remove_parent` or `remove_tx_without_depdendecies` and check that tx doesn't have dependencies that require an update.

Reply:
This will also be used in the remove of committed txs implemented here : https://github.com/FuelLabs/fuel-core/pull/2216/files#diff-e3dd0aa4d91212e38085e1deb6b3eb457e12df2a9d7b18de9f98125eabcfb4e2R194. But they should also have no dependencies otherwise I will add something to drop them, so I changed the name and added verification. Ty